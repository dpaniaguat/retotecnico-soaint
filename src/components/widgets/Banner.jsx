import React from 'react';
import './sass/Banner.scss';

export const Banner = () => {
    return (
        <div className="banner">
            <img className="banner__imagen" src="https://i.pinimg.com/originals/c3/d2/5d/c3d25dba7a9ae1a7f900d51631d9217f.png" alt="banner principal" />
        </div>
    )
}
