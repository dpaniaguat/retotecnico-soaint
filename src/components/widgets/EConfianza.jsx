// import "./sass/Widget.scss";
import "./sass/EConfianza.scss";
import confianza_1 from '../../assets/images/cards.webp';
import confianza_2 from '../../assets/images/delivery.webp';
import confianza_3 from '../../assets/images/security.webp';

export const EConfianza = () => {
  return (
   <>
   <div className="widget-econfianza">
    <div className="confianza-elements">
        <div className="confianza-elements__item">
          <img src={confianza_1} alt="" />
          <span>Las mejores ofertas</span>
        </div>
        <div className="confianza-elements__item">
          <img src={confianza_2} alt="" />
          <span>Programar es divertido y seguro!</span>
        </div>
        <div className="confianza-elements__item">
          <img src={confianza_3} alt="" />
          <span>Somos expertos en ReactJS</span>
        </div>
        </div>
      </div>
   </>
  );
};
