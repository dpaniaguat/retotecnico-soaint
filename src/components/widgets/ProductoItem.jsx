import React, { useContext } from "react";
import { Link } from "react-router-dom";
import CartContext from "../../context/cart/CartContext";
import { agregarProducto } from "../../utils/appCart";
import { parseMoney } from "../../utils/util";

export const ProductoItem = ({ id, title, price, image }) => {
  const { addToCart, showHideCart } = useContext(CartContext);

  const handleAddToCart = (id, title, price, image) => {
    const producto = {
      id,
      titulo: title,
      precio: price,
      imagen: image,
      cantidad: 1,
    };
    const productosLS = agregarProducto(producto);

    addToCart(productosLS);

    /* 
    desactivar en caso no requiera mostrar el resumen del carrito 
    cada vez que se a√±ade un producto
    */
    showHideCart();
  };

  return (
    <div className="producto" key={id}>
      <Link exact to={`/producto/${id}`}>
        <div className="producto-imagen">
          <img src={image} alt={title} />
        </div>
        <div className="producto-info">
          <h4 className="producto-info__title" title={title}>
            {title.toString().length > 25
              ? `${title.toString().slice(0, 20)}...`
              : title}
          </h4>
          <p className="producto-info__precios">
            <span>S/{parseMoney(price)}</span>
          </p>
        </div>
      </Link>
      <button
        className="producto-info__btnAddToCart"
        onClick={() => handleAddToCart(id, title, price, image)}
      >
        Agregar al carrito
      </button>
    </div>
  );
};
