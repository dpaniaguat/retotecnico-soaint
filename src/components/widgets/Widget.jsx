import "./sass/Widget.scss";

const Widget = ({ title = "", subtitle = "", children }) => {
  return (
    <section className="widget">
      {title.length > 0 && (
        <div className="widget-head">
          <h3 className="widget-head__title">{title}</h3>
          {subtitle.length > 0 && (
            <span className="widget-head__subtitle">{subtitle}</span>
          )}
          <hr />
        </div>
      )}
      <div className="widget-body">{children}</div>
    </section>
  );
};

export default Widget;
