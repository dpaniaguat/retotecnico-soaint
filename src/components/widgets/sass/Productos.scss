.widget-productos {
    display           : flex;
    flex-flow         : column wrap;
    justify-content   : center;
    align-items       : center;
    // width          : 80vw;

    h2 {
        margin-top: 2rem;
    }

    .productos {
        display                 : grid;
        grid-gap                : 7rem;
        gap                     : 11rem;
        grid-auto-rows          : 30rem;
        /* grid-template-columns: repeat(auto-fill,minmax(15rem,1fr)); */
        grid-template-columns   : repeat(auto-fill, minmax(min(100%, 15rem), 1fr));
        padding                 : 30px;

        .producto {
            height          : 40rem;
            padding         : 10px;
            border          : 1px solid #ccc;
            background-color: white;
            border-radius   : 10px;
            cursor          : pointer;
            display         : flex;
            flex-flow       : column wrap;
            align-items     : center;
            justify-content : center;

            a {
                text-decoration: none;
                color          : rgb(97, 93, 93);
                // height: 30rem;
            }

            &-imagen {
                display            : flex;
                flex-flow          : column wrap;
                align-items        : center;
                justify-content    : center;
                //  height         : 14rem;

                img {
                    width     : 60%;
                    box-sizing: border-box;
                }
            }

            &:hover {
                background-color: #f5f5f5;
                transition      : background-color ease-in .4s;
            }

            .producto-info {
                &__title {
                    height         : 100px;
                    border-bottom  : 2px solid #ccc;
                    margin-bottom  : 10px;
                    padding-bottom : 10px;
                    display        : flex;
                    align-items    : center;
                    justify-content: center;
                }

                &__precios {
                    display       : flex;
                    flex-direction: column;
                    font-weight   : bold;
                    align-items   : flex-end;
                    margin-bottom : 20px;
                }

                &__btnAddToCart {
                    width           : 25rem;
                    color           : white;
                    background-color: #005e85;
                    height          : 2rem;
                    border-radius   : 8px;
                    border-color    : transparent;
                    text-transform  : uppercase;
                    cursor          : pointer;
                    font-size       : 0.7rem;
                }
            }
        }
    }
}

@media(min-width:650px) {
    .widget-productos {
        .productos {
            display                 : grid;
            grid-gap                : 7rem;
            grid-gap                : 2rem;
            gap                     : 1rem;
            grid-auto-rows          : 30rem;
            /* grid-template-columns: repeat(auto-fill,minmax(15rem,1fr)); */
            grid-template-columns   : repeat(auto-fill, minmax(min(100%, 15rem), 1fr));
            padding                 : 30px;

            .producto {
                height          : 30rem;
                padding         : 10px;
                border          : 1px solid #ccc;
                background-color: white;
                border-radius   : 10px;
                cursor          : pointer;
                display         : flex;
                flex-flow       : column wrap;
                align-items     : center;
                justify-content : center;

                &-imagen {
                    height: 14rem;
                }

                .producto-info__btnAddToCart {
                    width: 14rem;
                }
            }
        }
    }

}

@media(min-width:900px) {
    .widget-productos {
        .productos {
            display              : grid;
            gap                  : 2rem;
            grid-auto-rows       : 30rem;
            grid-template-columns: repeat(auto-fill, minmax(min(100%, 15rem), 1fr));
            padding              : 35px;

            .producto {
                height          : 30rem;
                padding         : 10px;
                border          : 1px solid #ccc;
                background-color: white;
                border-radius   : 10px;
                cursor          : pointer;
                display         : flex;
                flex-flow       : column wrap;
                align-items     : center;
                justify-content : center;

                &-imagen {
                    height: 14rem;
                }

                .producto-info__btnAddToCart {
                    width: 14rem;
                }
            }
        }
    }

}