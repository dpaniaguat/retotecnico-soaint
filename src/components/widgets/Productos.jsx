import React from "react";
import { useFetch } from "../../hooks/useFetch";
import Loading from "../common/Loading";
import { ProductoItem } from "./ProductoItem";
import "./sass/Productos.scss";

export const Productos = ({ viewHeaderCounter = false }) => {
  const { loading, data } = useFetch(process.env.REACT_APP_API_URL);

  return (
    <section className="widget-productos">
      {viewHeaderCounter && (
        <div className="header-count">
          {data && <h2>Mostrando {data.length} Productos</h2>}
        </div>
      )}

      <div className="productos">
        {loading && <Loading texto="Cargando productos..." />}
        {data && data.map((item) => <ProductoItem key={item.id} {...item} />)}
      </div>
    </section>
  );
};
