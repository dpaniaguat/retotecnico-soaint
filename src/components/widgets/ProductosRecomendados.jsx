import React from "react";
import { useFetch } from "../../hooks/useFetch";
import Loading from "../common/Loading";
import { ProductoItem } from "./ProductoItem";
import "./sass/ProductosRecomendados.scss";

export const ProductosRecomendados = () => {
  
  const { loading, data } = useFetch( process.env.REACT_APP_API_URL);
  
  return (
    <section className="widget-productos-recomendados">
      <h3>Productos recomendados</h3>
      <div className="productos">
        {loading && <Loading texto="Cargando productos..." />}
        {data&& (data
          .sort(() => Math.random() - Math.random())
          .slice(0, 5)
          .map((data) => (
            <ProductoItem key={data.id} {...data} />
          )))}
      </div>
    </section>
  );
};
