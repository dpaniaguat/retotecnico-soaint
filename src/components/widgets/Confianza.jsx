import "./sass/Widget.scss";
import "./sass/Confianza.scss";
import confianza_1 from '../../assets/images/confianza_1.png';
import confianza_2 from '../../assets/images/confianza_2.png';
import confianza_3 from '../../assets/images/confianza_3.png';

export const Confianza = () => {
  return (
   <>
   <div className="widget__elements confianza">
        <div className="widget__elements--item">
          <img src={confianza_1} alt="" />
          <span>Las mejores ofertas</span>
        </div>
        <div className="widget__elements--item">
          <img src={confianza_2} alt="" />
          <span>Programar es divertido y seguro!</span>
        </div>
        <div className="widget__elements--item">
          <img src={confianza_3} alt="" />
          <span>Somos expertos en ReactJS</span>
        </div>
      </div>
   </>
  );
};
