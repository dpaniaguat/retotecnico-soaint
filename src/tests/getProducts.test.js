import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { Productos } from '../components/widgets/Productos'

configure({ adapter: new Adapter() });

test('message box', () => {
    const wrapper = shallow(<Productos viewHeaderCounter={true} />);
    expect(wrapper).toMatchSnapshot();
})



