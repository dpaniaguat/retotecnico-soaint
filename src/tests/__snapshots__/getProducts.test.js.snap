// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message box 1`] = `ShallowWrapper {}`;
